src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs

```diff
@@ -268,8 +270,60 @@ public Task<IEnumerable<ReferenceMethodDescriptor>> FindReferenceMethodsAsync(So
             using (solution.Services.CacheService?.EnableCaching(document.Project.Id))
             {
                 var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var declaredSymbol = semanticModel.GetDeclaredSymbol(syntaxNode, cancellationToken);
 
-                return semanticModel.GetDeclaredSymbol(syntaxNode, cancellationToken)?.ToDisplayString(MethodDisplayFormat);
+                if (declaredSymbol == null)
+                {
+                    return string.Empty;
+                }
+
+                var parts = declaredSymbol.ToDisplayParts(MethodDisplayFormat);
+                var pool = PooledStringBuilder.GetInstance();
+
+                try
+                {
+                    var actualBuilder = pool.Builder;
+                    var previousWasClass = false;
+
+                    for (var index = 0; index < parts.Length; index++)
+                    {
+                        var part = parts[index];
+                        if (previousWasClass && 
+                            part.Kind == SymbolDisplayPartKind.Punctuation &&
+                            index < parts.Length - 1)
+                        {
+                            switch (parts[index + 1].Kind)
+                            {
+                                case SymbolDisplayPartKind.ClassName:
+                                case SymbolDisplayPartKind.DelegateName:
+                                case SymbolDisplayPartKind.EnumName:
+                                case SymbolDisplayPartKind.ErrorTypeName:
+                                case SymbolDisplayPartKind.InterfaceName:
+                                case SymbolDisplayPartKind.StructName:
+                                    actualBuilder.Append('+');
+                                    break;
+
+                                default:
+                                    actualBuilder.Append(part);
+                                    break;
+                            }
+                        }
+                        else
+                        {
+                            actualBuilder.Append(part);
+                        }
+
+                        previousWasClass = part.Kind == SymbolDisplayPartKind.ClassName ||
+                                           part.Kind == SymbolDisplayPartKind.InterfaceName ||
+                                           part.Kind == SymbolDisplayPartKind.StructName;
```

Comment:
It it worth also having the same 6 values as you use above for any error scenarios?